core/src/SYCL/Kokkos_SYCL_Team.hpp

```diff
@@ -364,12 +366,13 @@ class SYCLTeamMember {
   // Private for the driver
 
   KOKKOS_INLINE_FUNCTION
-  SYCLTeamMember(void* shared, const int shared_begin, const int shared_size,
-                 void* scratch_level_1_ptr, const int scratch_level_1_size,
-                 const sycl::nd_item<2> item)
+  SYCLTeamMember(sycl::local_ptr<void> shared, const int shared_begin,
+                 const int shared_size,
+                 sycl::global_ptr<void> scratch_level_1_ptr,
```

Comment:
Did you mean to commit that change?

Reply:
Yes, this seems to fit better here than in #4396.